src/coreclr/jit/gcencode.cpp

```diff
@@ -4065,13 +4065,7 @@ void GCInfo::gcMakeRegPtrTable(
 {
     GCENCODER_WITH_LOGGING(gcInfoEncoderWithLog, gcInfoEncoder);
 
-    // TODO: Decide on whether we should enable this optimization for all
-    // targets: https://github.com/dotnet/runtime/issues/103917
-#ifdef TARGET_XARCH
     const bool noTrackedGCSlots = compiler->opts.MinOpts() && !compiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_PREJIT);
```

Comment:
@jkotas do you know if there is any dependency on the call site locations by crossgen2? I'm not sure if the `PREJIT` condition can also be removed here.

Reply:
I do not think crossgen2 has any dependency on that. NativeAOT might have (unintentional) dependency on it - you may want to run NativeAOT outer loop.

The primary motivation for this feature was JIT throughput improvement. This condition is likely result of that thinking - spending the extra cycles for AOT is no big deal.

I do not see a problem with removing this condition.

Reply:
The change in https://github.com/dotnet/runtime/pull/102680/files#r1653150895  may make us sensitive to missing safepoints. 

Mostly just a matter of changing asserts though. We know the caller site is GC-safe. (since we are in a GC-capable callee)

Reply:
Do you want me to revert the `PREJIT` change for now?